src/featuredetection/regexp.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module utils/featuredetection/regexp
+ */
+
+/**
+ * Holds feature detection related to regular expressions used by the editor.
+ *
+ * @protected
+ * @namespace
+ */
+export default {
+	/**
+	 * Indicates whether the current browser supports ES2018 Unicode properties like `\p{P}` or `\p{L}`.
+	 * More information about unicode properties might be found [here](https://www.unicode.org/reports/tr44/#GC_Values_Table).
+	 *
+	 * @type {Boolean}
+	 */
+	isUnicodePropertySupported: ( function() {
+		let isSupported = false;
+
+		// Feature detection for Unicode properties. Added in ES2018. Currently Firefox and Edge do not support it.
+		// See https://github.com/ckeditor/ckeditor5-mention/issues/44#issuecomment-487002174.
+
+		try {
+			isSupported = 'ć'.search( new RegExp( '[\\p{L}]', 'u' ) ) === 0;
```

Comment:
Why `new RegExp` and not a literal?

Reply:
@Reinmar I'm investigating this - ESLint is complaining about this.

Reply:
I mean that we have to bumpt the ESLint rules used in CKEditor to be able to use `/\p{L}/u` AFAICS.

Reply:
OH we've already published the changes and after updating dependencies it works :) So our ESLint config works with this now :tada:

Reply:
Ok so I tried it, and it's not working with our environment :(
Event when ESLint does not return an error, there is run `acorn` somewhere under the hood which throws a bunch of errors with those regexp literals:
<img width="1278" alt="Screenshot 2019-07-18 at 11 21 22" src="https://user-images.githubusercontent.com/9017322/61446324-36e50500-a94f-11e9-8685-7cc33f20e62f.png">

Reply:
OK, fair enough. Please leave a comment in the code with an explanation.